ouroboros-network/src/Ouroboros/Network/BlockFetch/ClientRegistry.hs

```diff
@@ -69,16 +74,18 @@ newFetchClientRegistry = FetchClientRegistry <$> newEmptyTMVarIO
 --
 -- It also manages synchronisation with the corresponding chain sync client.
 --
-bracketFetchClient :: forall m a peer header block.
+bracketFetchClient :: forall m a peer header block version.
                       (MonadThrow m, MonadSTM m, MonadFork m, MonadMask m,
                        Ord peer)
                    => FetchClientRegistry peer header block m
-                   -> NodeToNodeVersion
+                   -> version
+                   -> (version -> Bool) -- ^ is pipelining enabled function
```

Comment:
We could pass in `WhetherReceivingTentativeBlocks` here directly (or `version -> ...`) instead of `version -> Bool` for more instructive types.

Reply:
Thank you @amesgen I will look into it!

Reply:
I liked this suggestion, if you wouldn't mind re-reviewing now that I addressed it :)

Reply:
Looks nice! Just a thought (feel free to ignore, I don't think it is too important): It might make sense to move `WhetherReceivingTentativeBlocks` next to `isPipeliningEnabled` (and then rename it in order not to stress the "receiving" aspect which was/is appropriate in the BlockFetch client context), as it is now also used for the ChainSync server:

https://github.com/input-output-hk/ouroboros-network/blob/d87bce3968157d48489795191dbde8ebd4739613/ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs#L537-L545